commit e80c52986e1bb3afa6a92c58b1cb897877923a66
Author: Tom Stellard <tstellar@redhat.com>
Date:   Tue Feb 1 23:13:01 2022 -0800

    [docs] Remove hard-coded version numbers from sphinx configs
    
    This updates all the non-runtime project release notes to use the
    version number from CMake instead of the hard-coded version numbers
    in conf.py.
    
    It also hides warnings about pre-releases when the git suffix
    is dropped from the LLVM version in CMake.
    
    Reviewed By: MaskRay
    
    Differential Revision: https://reviews.llvm.org/D112181

Index: llvm-toolchain-snapshot_20~++20240920082629+ea578804c81b/clang-tools-extra/docs/conf.py
===================================================================
--- llvm-toolchain-snapshot_20~++20240920082629+ea578804c81b.orig/clang-tools-extra/docs/conf.py
+++ llvm-toolchain-snapshot_20~++20240920082629+ea578804c81b/clang-tools-extra/docs/conf.py
@@ -44,6 +44,15 @@ master_doc = "index"
 project = "Extra Clang Tools"
 copyright = "2007-%d, The Clang Team" % date.today().year
 
+# The version info for the project you're documenting, acts as replacement for
+# |version| and |release|, also used in various other places throughout the
+# built documents.
+#
+# The short version.
+version = '15'
+# The full version, including alpha/beta/rc tags.
+release = '15'
+
 # The language for content autogenerated by Sphinx. Refer to documentation
 # for a list of supported languages.
 # language = None
@@ -78,11 +87,6 @@ pygments_style = "friendly"
 # A list of ignored prefixes for module index sorting.
 # modindex_common_prefix = []
 
-in_progress_title = "(In-Progress) " if tags.has("PreRelease") else ""
-
-rst_epilog = f"""
-.. |ReleaseNotesTitle| replace:: {in_progress_title} Release Notes
-"""
 
 # -- Options for HTML output ---------------------------------------------------
 
Index: llvm-toolchain-snapshot_20~++20240920082629+ea578804c81b/clang/docs/conf.py
===================================================================
--- llvm-toolchain-snapshot_20~++20240920082629+ea578804c81b.orig/clang/docs/conf.py
+++ llvm-toolchain-snapshot_20~++20240920082629+ea578804c81b/clang/docs/conf.py
@@ -57,6 +57,15 @@ master_doc = "index"
 project = "Clang"
 copyright = "2007-%d, The Clang Team" % date.today().year
 
+# The version info for the project you're documenting, acts as replacement for
+# |version| and |release|, also used in various other places throughout the
+# built documents.
+#
+# The short version.
+version = '15'
+# The full version, including alpha/beta/rc tags.
+release = '15'
+
 # The language for content autogenerated by Sphinx. Refer to documentation
 # for a list of supported languages.
 # language = None
@@ -88,11 +97,6 @@ exclude_patterns = ["_build"]
 # The name of the Pygments (syntax highlighting) style to use.
 pygments_style = "friendly"
 
-in_progress_title = "(In-Progress) " if tags.has("PreRelease") else ""
-
-rst_epilog = f"""
-.. |ReleaseNotesTitle| replace:: {in_progress_title} Release Notes
-"""
 
 # -- Options for HTML output ---------------------------------------------------
 
Index: llvm-toolchain-snapshot_20~++20240920082629+ea578804c81b/lld/docs/conf.py
===================================================================
--- llvm-toolchain-snapshot_20~++20240920082629+ea578804c81b.orig/lld/docs/conf.py
+++ llvm-toolchain-snapshot_20~++20240920082629+ea578804c81b/lld/docs/conf.py
@@ -43,6 +43,15 @@ master_doc = "index"
 project = "lld"
 copyright = "2011-%d, LLVM Project" % date.today().year
 
+# The version info for the project you're documenting, acts as replacement for
+# |version| and |release|, also used in various other places throughout the
+# built documents.
+#
+# The short version.
+version = '15'
+# The full version, including alpha/beta/rc tags.
+release = '15'
+
 # The language for content autogenerated by Sphinx. Refer to documentation
 # for a list of supported languages.
 # language = None
Index: llvm-toolchain-snapshot_20~++20240920082629+ea578804c81b/llvm/cmake/modules/AddSphinxTarget.cmake
===================================================================
--- llvm-toolchain-snapshot_20~++20240920082629+ea578804c81b.orig/llvm/cmake/modules/AddSphinxTarget.cmake
+++ llvm-toolchain-snapshot_20~++20240920082629+ea578804c81b/llvm/cmake/modules/AddSphinxTarget.cmake
@@ -40,10 +40,6 @@ function (add_sphinx_target builder proj
     set(ARG_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}")
   endif()
 
-  if ("${LLVM_VERSION_SUFFIX}" STREQUAL "git")
-    set(PreReleaseTag "-tPreRelease")
-  endif()
-
   add_custom_target(${SPHINX_TARGET_NAME}
                     COMMAND ${CMAKE_COMMAND} -E env ${ARG_ENV_VARS}
                             ${SPHINX_EXECUTABLE}
@@ -51,9 +47,6 @@ function (add_sphinx_target builder proj
                             -d "${SPHINX_DOC_TREE_DIR}"
                             -q # Quiet: no output other than errors and warnings.
                             -t builder-${builder} # tag for builder
-                            -D version=${LLVM_VERSION_MAJOR}
-                            -D release=${PACKAGE_VERSION}
-                            ${PreReleaseTag}
                             ${SPHINX_WARNINGS_AS_ERRORS_FLAG} # Treat warnings as errors if requested
                             "${ARG_SOURCE_DIR}" # Source
                             "${SPHINX_BUILD_DIR}" # Output
Index: llvm-toolchain-snapshot_20~++20240920082629+ea578804c81b/llvm/docs/conf.py
===================================================================
--- llvm-toolchain-snapshot_20~++20240920082629+ea578804c81b.orig/llvm/docs/conf.py
+++ llvm-toolchain-snapshot_20~++20240920082629+ea578804c81b/llvm/docs/conf.py
@@ -61,6 +61,15 @@ master_doc = "index"
 project = "LLVM"
 copyright = "2003-%d, LLVM Project" % date.today().year
 
+# The version info for the project you're documenting, acts as replacement for
+# |version| and |release|, also used in various other places throughout the
+# built documents.
+#
+# The short version.
+version = '15'
+# The full version, including alpha/beta/rc tags.
+release = '15'
+
 # The language for content autogenerated by Sphinx. Refer to documentation
 # for a list of supported languages.
 # language = None
Index: llvm-toolchain-snapshot_20~++20240920082629+ea578804c81b/polly/docs/conf.py
===================================================================
--- llvm-toolchain-snapshot_20~++20240920082629+ea578804c81b.orig/polly/docs/conf.py
+++ llvm-toolchain-snapshot_20~++20240920082629+ea578804c81b/polly/docs/conf.py
@@ -44,6 +44,15 @@ master_doc = "index"
 project = "Polly"
 copyright = "2010-%d, The Polly Team" % date.today().year
 
+# The version info for the project you're documenting, acts as replacement for
+# |version| and |release|, also used in various other places throughout the
+# built documents.
+#
+# The short X.Y version.
+version = '15.0'
+# The full version, including alpha/beta/rc tags.
+release = '15.0'
+
 # The language for content autogenerated by Sphinx. Refer to documentation
 # for a list of supported languages.
 # language = None
@@ -78,11 +87,6 @@ pygments_style = "friendly"
 # A list of ignored prefixes for module index sorting.
 # modindex_common_prefix = []
 
-in_progress_title = "(In-Progress) " if tags.has("PreRelease") else ""
-
-rst_epilog = f"""
-.. |ReleaseNotesTitle| replace:: {in_progress_title} Release Notes
-"""
 
 # -- Options for HTML output ---------------------------------------------------
 
